PAY-4797: Adds new page and diagram for the paypal vault payment method (#250)
AdobeDocs/commerce-webapi: src/pages/graphql/payment-methods/payment-services-paypal-vault.md

---
title: Payment Services Paypal Vault payment method
description: Learn how to use the GraphQL API mutation for the Payment Services Paypal Vault payment solution.
keywords:
  - GraphQL
  - Payments
---

> # Payment Services Paypal Vault payment method

Reviewer: ```suggestion
# Payment Services Vault payment method
```

> Payment Services Paypal Vault is a payment method where a stored card is used for the transaction.

Reviewer 2: ```suggestion
The [Payment Services Vault payment method](https://experienceleague.adobe.com/docs/commerce-merchant-services/payment-services/payments-checkout/vaulting.html) accesses credit card credentials that the customer has previously saved—or “vaulted”.
```

Author: we have a conflicting suggestion with Shrie's here: https://github.com/AdobeDocs/commerce-webapi/pull/250#discussion_r1410843438

The following conditions must be true to use this payment method:

> -  The shopper must be a logged-in customer.

Reviewer 2: ```suggestion
-  The customer must be logged-in.
```

I only use the word shopper in the GraphQL context when it's possible that the person can be a logged-in customer or a guest.

-  The customer must have previously saved their payment information in the vault.

You cannot use this payment method if the customer decides to use a credit or debit card that is not stored in the vault.

## Payment Services Paypal Vault workflow

The following diagram shows the workflow for placing an order when Payflow Pro Vault is the selected payment method.

> ![Payment Services Paypal Vault sequence diagram](../../_images/graphql/payment-services-paypal-vault.svg)

Reviewer: ```suggestion
![Payment Services Vault sequence diagram](../../_images/graphql/payment-services-paypal-vault.svg)
```

Reviewer: Take "paypal" out of the extension name + here

1. Use the [`customerPaymentTokens`](../schema/checkout/queries/customer-payment-tokens.md) query to retrieve the payment tokens the customer has stored in the vault.

1. Commerce returns an array of payment tokens.

1. The client integration renders the token information, and the customer selects a payment method.

   When the customer selects a stored payment method, the client integration uses the [`setPaymentMethodOnCart`](../schema/cart/mutations/set-payment-method.md) mutation to set the payment method to `payment_services_paypal_vault`.

1. Commerce returns a `Cart` object.

1. Run `placeOrder`.

1.  Commerce sends an authorization request to PayPal.

> 1.  PayPal returns the result to Commerce.

Reviewer: ```suggestion
1. PayPal returns the result to Commerce.
```

> 1.  Commerce creates an order.

Reviewer: ```suggestion
1. Commerce creates an order.
```

## Additional Payment information

> When you set the payment method to Payment Services Paypal Vault in the [`setPaymentMethodOnCart`](../schema/cart/mutations/set-payment-method.md) mutation, this mutation has an extra field, `public_hash`, which identifies the stored card being used.

Reviewer: ```suggestion
When you set the payment method to Payment Services Vault in the [`setPaymentMethodOnCart`](../schema/cart/mutations/set-payment-method.md) mutation, this mutation has an extra field, `public_hash`, which identifies the stored card being used.
```

> When you set the payment method to Payment Services Paypal Vault in the [`setPaymentMethodOnCart`](../schema/cart/mutations/set-payment-method.md) mutation, this mutation has an extra field, `public_hash`, which identifies the stored card being used.

Reviewer 2: ```suggestion
When you set the payment method to Payment Services Vault in the [`setPaymentMethodOnCart`](../schema/cart/mutations/set-payment-method.md) mutation, you must supply a value for the `public_hash` field. Run the [`customerPaymentTokens`](../schema/checkout/queries/customer-payment-tokens.md) query to retrieve the value.
```

> This `public_hash` comes from the [`customerPaymentTokens`](../schema/checkout/queries/customer-payment-tokens.md) query:

Reviewer 2: No need for the code sample. Delete lines 49-62, plus any extra whitespace

```graphql
{
    customerPaymentTokens {
        items {
            public_hash
            details
            payment_method_code
            type
        }
    }
}
```

### `payment_services_paypal_vault` attributes

The `setPaymentMethodOnCart` mutation has a `payment_services_paypal_vault` object that contains the following attributes:

Attribute |  Data Type | Description
--- | --- | ---

> `location` | PaymentLocation | The origin location for that payment request. The possible values are PRODUCT_DETAIL, MINICART, CART, CHECKOUT, ADMIN

Reviewer 2: There's no location. Delete

`payments_order_id` | String | The unique order ID generated in Commerce if Payment Services is enabled after PayPal returns the `paypal_order_id`
`paypal_order_id` | String | The unique order ID generated by PayPal when receiving the authorization request
`payment_source` | String | The identifiable payment source for the payment method. The possible values are credit card, PayPal, and Apple Pay
`public_hash` | String | The public hash of the payment token

### Example usage

> The following example shows the `setPaymentMethodOnCart` mutation constructed for the Payment Services Paypal Vault payment method.

Reviewer: ```suggestion
The following example shows the `setPaymentMethodOnCart` mutation constructed for the Payment Services Vault payment method.
```

**Request:**

```graphql
mutation {
    setPaymentMethodOnCart ( input: {
        cart_id: "uocGxUi5H97XFAMhY3s66q4aFYG3Bmdr",
        payment_method: {
            code: "payment_services_paypal_vault",
            payment_services_paypal_vault: {
                payment_source: "vault",
                payments_order_id: "mp-order-a4babd34-13d3-4ac0-b1b0-109bb7be1574",
                paypal_order_id: "9R90936863877801D",
                public_hash: "shew83bdbd83y2bdjbwb2ib2ds"

>                 location: PRODUCT_DETAIL

Reviewer 2: Delete this line
